src/ucp/wireup/wireup.c

```diff
@@ -1564,6 +1564,7 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, unsigned ep_init_flags,
     /* establish connections on all underlying endpoints */
     for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
         if (ucp_ep_get_cm_lane(ep) == lane) {
+            ucp_wireup_cm_ep_pending_extract(lane, ep, &replay_pending_queue);
```

Comment:
IMO need to extract `replay_pending_queue`  logic from `ucp_wireup_check_config_intersect` function, extract all pending requests from all lanes before it and replay them in the end.

Reply:
Creating one function that pop all the elements from the pending queue seems like a good idea in theory. This way this logic will be in just one place instead of 3+ places in the code.
The downside is that it's will require changing/removing logic from existing code (`ucp_wireup_check_config_intersect`).
It's will require more time and it also can cause issues with "Old proto".

I can try to do it and see how it's works.
WDYT?

Reply:
@evgeny-leksikov i'm not sure it's possible to do the refactoring before ProtoV2 support for stream is added.
If we have pending requests with stream protocol in the pending queue a common function for extracting the requests will not work when ProtoV2 is enabled if i'm not mistaken.

Maybe it's better to postpone this, at least until ProtoV2 support all protocols?

Reply:
maybe if it's hard to split paths inside of the function, we can implement different independent functions for protov1 and protov2?

Reply:
We can do that but it's will cause new code duplication and new branches in the code.
If the main goal of this suggested refactoring is to reduce code duplication i'm not sure if we want to go in this path.
Eventually we will be left only with the ProtoV2 functions, The questions is if we want to do it now in this PR or wait until Old proto is deprecated.

Let me know what you think.

Reply:
If it's over-complicating and the profit is unclear, maybe leave it as is for now? @evgeny-leksikov WDYT?